src/main/java/jenkins/branch/MultiBranchProjectDisplayNamingStrategy.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * The MIT License
+ *
+ * Copyright (c) 2021, Christoph Obexer.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package jenkins.branch;
+
+import edu.umd.cs.findbugs.annotations.NonNull;
+import jenkins.scm.api.metadata.ObjectMetadataAction;
+
+import static java.lang.String.format;
+import static org.apache.commons.lang.StringUtils.isBlank;
+
+/**
+ * Possible Display naming strategies.
+ *
+ * @since 2.7.1
+ */
+public enum MultiBranchProjectDisplayNamingStrategy {
+    /**
+     * Use the name as specified by the scm connector. (traditional behaviour)
+     */
+    RAW(false) {
+        @Override
+        public String generateName(@NonNull final String rawName, final String displayName) {
+            return rawName;
+        }
+    },
```

Comment:
why would you have this.  this is the behavior when the trait does not exist - so to get this behaviour just delete the trait?

Reply:
That's an interesting take.
Didn't consider it and never even played with it in the UI so I didn't really realize what it feels like using it. 
Will try to do that :+1:

Reply:
See [my other comment](https://github.com/jenkinsci/branch-api-plugin/pull/374#discussion_r1198644264) let's continue the discussion there.